cpp/autosar/src/rules/A12-4-1/DestructorOfABaseClassNotPublicVirtual.ql

```diff
@@ -29,9 +29,9 @@ predicate isProtectedNonVirtual(Destructor d) { d.isProtected() and not d.isVirt
 from Destructor d
 where
   not isExcluded(d, VirtualFunctionsPackage::destructorOfABaseClassNotPublicVirtualQuery()) and
-  isPossibleBaseClass(d.getDeclaringType(), _) and
+  d.getDeclaringType() instanceof BaseClass and
```

Comment:
can we add a testcase for this one also?

for some reason I actually cant replicate an FN on this:
```
template <class T1> class BaseClass7 {
public:
  ~BaseClass7() {}         // NON_COMPLIANT
};

template <class T>
class DerivedClass7 // COMPLIANT - not a base class itself
    : public BaseClass7<T> {
public:
  ~DerivedClass7() {}
};

void test() {
  BaseClass7<int> b;
  DerivedClass7<int> d;
}
```
I think it has to do with the fact that the previous implementation of `isPossibleBaseClass(d.getDeclaringType(), _)` (use a quick eval on that) actually does seem to be fine for the constructors/destructors for some reason but not the other member functions

if that is the case I do think it still does make sense to update this query also though

Reply:
That is because with a user-defined destructor, the compiler will not generate a copy/move constructor/operator.

That being said, adjusting your example did uncover 🚀 an issue that I have resolved.